consensus/broadcast/shardChainMessenger_test.go

```diff
@@ -512,3 +606,189 @@ func TestShardChainMessenger_BroadcastBlockDataLeaderShouldTriggerWaitingDelayed
+			return nil
+		}}
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	scm.PrepareBroadcastBlockDataValidator(nil, nil, nil, 1, pkBytes)
+}
+
+// This test verifies if delayedBlockBroadcaster.SetValidatorData is not called
+// Len of MiniBlocks is 0 and the execution stops
+func TestShardChainMessenger_PrepareBroadcastBlockDataValidatorShouldFailMiniBlocksLenZero(t *testing.T) {
+
+	pkBytes := make([]byte, 32)
+	miniBlocks := make(map[uint32][]byte)
+	headerMock := &testscommon.HeaderHandlerStub{}
+
+	args := createDefaultShardChainArgs()
+
+	args.DelayedBroadcaster = &mock.DelayedBroadcasterMock{
+		SetValidatorDataCalled: func(data *shared.DelayedBroadcastData) error {
+			require.Fail(t, "SetValidatorData should not be called")
+			return nil
+		}}
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	scm.PrepareBroadcastBlockDataValidator(headerMock, miniBlocks, nil, 1, pkBytes)
+}
+
+// This test verifies if delayedBlockBroadcaster.SetValidatorData is not called
+// CalculateHash should return error and the execution stops
+func TestShardChainMessenger_PrepareBroadcastBlockDataValidatorShouldFailCalculateHashErr(t *testing.T) {
+
+	pkBytes := make([]byte, 32)
+	miniBlocks := map[uint32][]byte{1: {}}
+	headerMock := &testscommon.HeaderHandlerStub{}
+
+	args := createDefaultShardChainArgs()
+
+	args.DelayedBroadcaster = &mock.DelayedBroadcasterMock{
+		SetValidatorDataCalled: func(data *shared.DelayedBroadcastData) error {
+			require.Fail(t, "SetValidatorData should not be called")
+			return nil
+		}}
+
+	marshallerErr := fmt.Errorf("marshalizer error")
+	args.Marshalizer = &testscommon.MarshallerStub{MarshalCalled: func(obj interface{}) ([]byte, error) {
+		return nil, marshallerErr
+	}}
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	scm.PrepareBroadcastBlockDataValidator(headerMock, miniBlocks, nil, 1, pkBytes)
+}
+
+// This test verifies if delayedBlockBroadcaster.SetValidatorData is called
+func TestShardChainMessenger_PrepareBroadcastBlockDataValidatorShouldWork(t *testing.T) {
+
+	pkBytes := make([]byte, 32)
+	miniBlocks := map[uint32][]byte{1: {}}
+	headerMock := &testscommon.HeaderHandlerStub{}
+
+	args := createDefaultShardChainArgs()
+
+	varSetValidatorDataCalled := false
+	args.DelayedBroadcaster = &mock.DelayedBroadcasterMock{
+		SetValidatorDataCalled: func(data *shared.DelayedBroadcastData) error {
+			varSetValidatorDataCalled = true
+			return nil
+		}}
+
+	args.Marshalizer = &testscommon.MarshallerStub{MarshalCalled: func(obj interface{}) ([]byte, error) {
+		return nil, nil
+	}}
+
+	args.Hasher = &testscommon.HasherStub{ComputeCalled: func(s string) []byte {
+		return nil
+	}}
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	scm.PrepareBroadcastBlockDataValidator(headerMock, miniBlocks, nil, 1, pkBytes)
+
+	assert.True(t, varSetValidatorDataCalled)
+}
+
+// This test increases the code coverage and verify if delayedBroadcaster.Call is called
```

Comment:
Can you remove the comments on the Test functions? 
I think the tests are clear enough.

Reply:
updated